swap/simulations_test.go

```diff
@@ -676,33 +683,60 @@ func TestBasicSwapSimulation(t *testing.T) {
 	log.Info("Simulation ended")
 }
 
-func waitForChequeProcessed(t *testing.T, ts *testService, counter metrics.Counter, lastCount int64) error {
+func waitForChequeProcessed(t *testing.T, backend *swapTestBackend, counter metrics.Counter, lastCount int64, p *Peer, expectedLastPayout uint64) error {
 	t.Helper()
 	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
 	defer cancel()
 
 	// we are going to wait for two things:
-	// * that a cheque has been processed
+	// * that a cheque has been processed and confirmed
 	// * that the other side actually did process the testMsgPrice message
 	//   (by checking that the message counter has been increased)
 	var wg sync.WaitGroup
-	wg.Add(2)
+	errs := []string{}
```

Comment:
This slice may be changed in three different goroutines. It is possible to have a data race on it. That can cause to miss some errors, even in this case it does not matter so much. I think that it should be protected to have the correct code.